src/addon.cpp

```diff
@@ -26,49 +27,58 @@ class SHA3Hash: public ObjectWrap {
 
 	static void
 	Initialize(Handle<Object> target) {
-		HandleScope scope;
-		Local<FunctionTemplate> t = FunctionTemplate::New(New);
+		NanScope();
```

Comment:
Invoking a constructor in this manner creates a temporary object which is destroyed at the end of the statement. This is essentially a no-op. Give the new object a name: `NanScope scope`. The destructor will then be invoked a the end of the enclosing block, as intended.